provisioning/hare_setup.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import argparse
+import asyncio
+import json
+import logging
+import shutil
+import subprocess
+import os
+import yaml
+
+from cortx.utils.product_features import unsupported_features
+
+def execute_command_and_get_output(cmd):
+        process = subprocess.Popen(cmd, stdin=subprocess.PIPE,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        encoding='utf8')
+        out, _ = process.communicate()
+        if process.returncode:
+            raise Exception('Error occured while executing command')
```

Comment:
Usually when an error happens, the details can be found in stderr. Can we add stderr to the exception?

like `raise Exception(f'Command {cmd} exited with error code {returncode}. Command output: {err}')

Reply:
...or at least log the error details with `logger.error()`. Just imagine, how would you debug an issue that happened in the field and how much of information would be sufficient to understand what was wrong.

Reply:
Done